datafusion/src/datasource/mod.rs

```diff
@@ -198,3 +200,27 @@ fn get_col_stats(
         })
         .collect()
 }
+
+fn flatten_schema(schema: &Schema) -> Vec<&Field> {
```

Comment:
In the current uses, it seems we only need the total number of fields (i.e. return a `usize`) instead of the field information.

It seems to me that would be a bit more simple / faster than first collecting all the fields if we don't need the results of that.

Reply:
I'll try simplifying it...

Reply:
@Dandandan Simplifying it is not a solution as we need to read/calculate the statistics for the nested fields too. We will be using the flattened fields when calling `summarize_min_max` function.

Reply:
@Dandandan I added the `total_number_of_fields` function besides the flatten one.